localnet/kubernetes/values-relayminer-2.yaml

```diff
@@ -2,10 +2,9 @@ config:
   signing_key_name: supplier2
```

Comment:
_tl;dr Can we simplify it? Let's discuss..._
---

@red-0ne As I'm looking at this, I have a bunch of questions:

1. Why do we need `server_type`?
	2. Is the `server_typr` the same as `backend_url`?
3. Why to we need `listen_address`?
	4. 	Is `listen_address` going to be the same as `publicly_exposed_endpoints` on localhost w/o the scheme?
5. There is redundancy in ports, schemes and URLs. 

You don't have to answer these questions, but it's confusing because:
1. I thought we aligned on remove `server_type`
2. `listen_address` feels redundant.

I want to emphasize that it's not something we should change, but it's still not blatantly obvious.

I look at https://pkg.go.dev/net/http#Request and found that:

```go
// For client requests, Host optionally overrides the Host
// header to send. If empty, the Request.Write method uses
// the value of URL.Host. Host may contain an international
// domain name.
Host [string](https://pkg.go.dev/builtin#string)
```	
And:

```go
// For client requests, the URL's Host specifies the server to
// connect to, while the Request's Host field optionally
// specifies the Host header value to send in the HTTP
// request.
URL *[url](https://pkg.go.dev/net/url).[URL](https://pkg.go.dev/net/url#URL)
```

And from our implementation:

```go
URL:  &url.URL{Scheme: scheme, Host: proxiedServices[proxyServeName].ListenAddress},
Host: proxiedServices[proxyServeName].Suppliers[serviceId].PubliclyExposedEndpoints[0]
```

Under what circumstance would we not be using `[0]`?

Reply:
`listen_address` is needed and indicates on which address the relay miner will be listening to relays on.
`server_type` is removed now and `listen_address` changed to `listen_url`, we infer the `server_type` from the url scheme component.

As of `Host` and `Url`, this is how its is intended to work. The request's host is different from the url's to exercise how the `RelayMiner` accepts requests forwarded from reverse proxies and situations where the `listen_url` address is not publicly exposed.

> Under what circumstance would we not be using [0]?

This is a test scenario where we exercise requests sent to the first `PubliclyExposedEndpoints` and handled by the server listening at `ListenAddress`.